net-vpn/i2p/i2p-2.1.0.ebuild

```diff
@@ -0,0 +1,199 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="test"
+
+inherit java-pkg-2 java-ant-2 systemd
+
+DESCRIPTION="A privacy-centric, anonymous network"
+HOMEPAGE="https://geti2p.net"
+SRC_URI="https://files.i2p-projekt.de/${PV}/i2psource_${PV}.tar.bz2"
+
+LICENSE="Apache-2.0 Artistic BSD CC-BY-2.5 CC-BY-3.0 CC-BY-SA-3.0 EPL-1.0 GPL-2 GPL-3 LGPL-2.1 LGPL-3 MIT public-domain WTFPL-2"
+SLOT="0"
+
+KEYWORDS="~amd64 ~x86"
+LANGS=(
+	ar az bg ca cs da de el en es es-AR et fa fi fr gl he hi hr hu id it ja ko ku mg nb nl nn pl pt pt-BR ro ru sk sl sq
+	sr sv tk tr uk vi zh zh-TW
+)
+IUSE="${LANGS[@]/#/l10n_}"
+
+CP_DEPEND="
+	dev-java/java-getopt:1
+	dev-java/java-service-wrapper:0
+	dev-java/jstl-api:0
+	dev-java/jstl:0
+	net-libs/nativebiginteger:0
+	sys-devel/gettext:0[java]
+	>=www-servers/tomcat-9.0.73:9
+"
+DEPEND="
+	${CP_DEPEND}
+	>=virtual/jdk-1.8:*
+	test? (
+		dev-java/ant-junit4:0
+		dev-java/hamcrest:0
+		dev-java/hamcrest-core:1.3
+		dev-java/junit:4
+		dev-java/mockito:4
+	)
+"
+RDEPEND="
+	${CP_DEPEND}
+	acct-user/i2p
+	acct-group/i2p
+	>=virtual/jre-1.8:*
+"
+
+EANT_BUILD_TARGET="preppkg-base"
+# no scala as depending on antlib.xml not installed by dev-lang/scala
+EANT_TEST_TARGET="junit.test"
+JAVA_ANT_ENCODING="UTF-8"
+
+DOCS=(README.md history.txt)
+
+PATCHES=(
+	"${FILESDIR}/${PV}-fix-getopt-classpath.patch"
+)
+
+JAVA_PKG_NO_CLEAN=(
+	"*/apps/jetty/jetty-distribution-*" # need to package jetty
+)
+
+src_prepare() {
+	# java-pkg-2_src_prepare doesn't apply our patches
+	eapply "${PATCHES[@]}"
+	java-pkg-2_src_prepare
+
+	# remove most bundled
+	java-pkg_clean
+	( cat >> override.properties || die 'set unbundled properties' ) <<- EOF
+		require.gettext=true
+		with-libgetopt-java=true
+		with-libjakarta-taglibs-standard-java=true
+		with-libtomcat9-java=true
+		# with-geoip-database=true needs https://github.com/maxmind/geoip-api-java
+		# with-libjetty9-java=true needs a jetty ebuild
+		# with-libjson-simple-java=true needs https://cliftonlabs.github.io/json-simple
+	EOF
+
+	# unbundle getopt
+	rm -r core/java/src/gnu/getopt || die 'unbundle java-getopt code'
+	local d=core/java/build
+	mkdir "$d" || die 'create core build dir'
+	java-pkg_jar-from --into "$d" java-getopt-1 gnu{.,-}getopt.jar
```

Comment:
Instead of all these `java-pkg_jar-from` I'd prefer `java-pkg_getjars` which for `getopt`  could look like:
```

	# unbundle getopt
	rm -r core/java/src/gnu/getopt || die 'unbundle java-getopt code'
	local d=core/java/build
	mkdir "$d" || die 'create core build dir'
#	java-pkg_jar-from --into "$d" java-getopt-1 gnu{.,-}getopt.jar
	local getoptjar="$(java-pkg_getjars java-getopt-1)"
	sed \
		-e "s:/usr/share/java/gnu-getopt.jar:${getoptjar}:" \
		-e "s:build/gnu-getopt.jar:${getoptjar}:" \
		-i core/java/build.xml || die "cannot work on build.xml"
```
With my poor understanding of `build.xml` I have no idea how those values could be overwritten with `override.properties`.

Reply:
hoo, with a bit of playing with `EANT_GENTOO_CLASSPATH_EXTRA` and rewriting all build files, I was able to remove all get/from-jars, thanks for pushing it!